store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/ParquetStore.java

```diff
@@ -0,0 +1,335 @@
+
+/**
+ *
+ */
+public class ParquetStore extends Store {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ParquetStore.class);
+    private static final Set<StoreTrait> TRAITS = new HashSet<>();
+    private HashMap<String, ArrayList<Tuple3<Object[], Object[], String>>> groupToIndex;
+    private SchemaUtils schemaUtils;
+    private FileSystem fs;
+    private long currentSnapshot;
+
+    static {
+        TRAITS.add(StoreTrait.INGEST_AGGREGATION);
+        TRAITS.add(StoreTrait.ORDERED);
+        TRAITS.add(StoreTrait.PRE_AGGREGATION_FILTERING);
+    }
+
+    public ParquetStore(final Schema schema, final StoreProperties properties) throws StoreException {
+        super();
+        initialise(schema, properties);
+    }
+
+    public ParquetStore() {
+        super();
+    }
+
+    @Override
+    protected SchemaOptimiser createSchemaOptimiser() {
+        return new SchemaOptimiser(new ParquetSerialisationFactory());
+    }
+
+    @Override
+    public void initialise(final Schema schema, final StoreProperties properties) throws StoreException {
+        super.initialise(schema, properties);
+        this.schemaUtils = new SchemaUtils(this.getSchema());
+        loadIndices();
+    }
+
+    public FileSystem getFS() throws StoreException {
+        if (this.fs == null) {
+            try {
+                this.fs = FileSystem.get(new Configuration());
+            } catch (IOException e) {
+                throw new StoreException("Could not connect to the file system", e);
+            }
+        }
+        return this.fs;
+    }
+
+    public SchemaUtils getSchemaUtils() {
+        if (this.schemaUtils == null) {
+            this.schemaUtils = new SchemaUtils(getSchema());
+        }
+        return this.schemaUtils;
+    }
+
+    public Set<StoreTrait> getTraits() {
+        return TRAITS;
+    }
+
+    public boolean isValidationRequired() {
+        return false;
+    }
+
+    @Override
+    protected Context createContext(final User user) {
+        if (user instanceof SparkUser) {
+            final SparkConf conf = ((SparkUser) user).getSparkSession().sparkContext().getConf();
+            final String sparkSerialiser = conf.get("spark.serializer", null);
+            if (sparkSerialiser == null || !sparkSerialiser.equals("org.apache.spark.serializer.KryoSerializer")) {
+                LOGGER.warn("For the best performance you should set the spark config 'spark.serializer' = 'org.apache.spark.serializer.KryoSerializer'");
+            }
+            final String sparkKryoRegistrator = conf.get("spark.kryo.registrator", null);
+            if (sparkKryoRegistrator == null || !sparkKryoRegistrator.equals("uk.gov.gchq.gaffer.spark.serialisation.kryo.Registrator")) {
+                LOGGER.warn("For the best performance you should set the spark config 'spark.kryo.registrator' = 'uk.gov.gchq.gaffer.spark.serialisation.kryo.Registrator'");
+            }
+            return super.createContext(user);
```

Comment:
perhaps this could be refactored to something like:

final SparkUser sparkUser;
if (user instanceof SparkUser) {
sparkUser = (SparkUser) user;
...
} else {
...
sparkUser = new SparkUser(user, spark);
}
return sparkUser;